src/spaces/grid.jl

```diff
@@ -205,6 +197,37 @@ function nearby_ids(pos::ValidPos, model::ABM{<:AbstractAgent,<:GridSpace}, r =
     Iterators.flatten((s[i...] for i in nn))
 end
 
+"""
+    nearby_ids(pos, model::ABM{<:GridSpace}, r::Vector{Tuple{Int,UnitRange{Int}}})
+
+Return an iterator of ids over specified dimensions of `space` with fine grained control
+of distances from `pos` using each value of `r` via the (dimension, range) pattern.
+
+Example, with a `GridSpace((100, 100, 10))`: `r = [(1, -5:5), (3, 0:2)]`.
```

Comment:
I'm not sure this is as clear as it could be. So in principle here you would search the 3rd dimension, from current level up to at most +2 levels, right? Perhaps we should discuss this more in the docstring and also crossref to the battle royale example?

Reply:
Certainly. Will do so tomorrow.

Reply:
This will be easier to extend once the battle example is updated, so I'll fix this in #359.